nifi-nar-bundles/nifi-extension-utils/nifi-record-utils/nifi-standard-record-utils/src/main/java/org/apache/nifi/csv/CSVValidators.java

```diff
@@ -25,15 +25,18 @@
 import java.util.Set;
 
 public class CSVValidators {
+    private static final Set<String> illegalChars = new HashSet<>();
 
-    public static class SingleCharacterValidator implements Validator {
-        private static final Set<String> illegalChars = new HashSet<>();
+    static {
+        illegalChars.add("\r");
+        illegalChars.add("\n");
+    }
 
-        static {
-            illegalChars.add("\r");
-            illegalChars.add("\n");
-        }
+    public static final Validator SINGLE_CHAR_VALIDATOR = createSingleCharValidator(false);
+
+    public static final Validator EMPTY_AND_SINGLE_CHAR_VALIDATOR = createSingleCharValidator(true);
```

Comment:
I recommend naming it EMPTY_OR_SINGLE_CHAR_VALIDATOR as it contains a logical OR in the implementation. When reading the code of CSVUtils, at `.addValidator(CSVValidators.EMPTY_AND_SINGLE_CHAR_VALIDATOR)` I got a little confused. My first impression was that the property is required to be empty and contain a single character at the same time.